source/types.ts

```diff
@@ -225,3 +225,11 @@ export interface QueryOptions {
 }
 
 export interface CallOptions extends MutationOptions, QueryOptions {}
+
+export interface EntityTypeMap {
+  FileComponent: {};
+}
+export type EntityType = keyof EntityTypeMap;
```

Comment:
Setting `EntityType` like this unfortunately makes it unable to be extended - it won't automatically catch the augmented types. I'm not even sure if it works if we straight up use `keyof EntityTypeMap` all the time. We have to investigate that.

Reply:
Aha, interesting. I did not know that. That's weird to me. I'll try to think of other ways to extend the types.

Reply:
I have now switched to `keyof` everywhere. If you get a chance, it'd be awesome to test it out. If you don't have much time, I'll see if I can try it soon too!

Reply:
Probably won't have time in the coming two weeks. Might sneak in an extra evening session, but don't expect a review before early October unfortunately.

Reply:
Sure thing! I'll see what I can do!